src/Generator.php

```diff
@@ -4,16 +4,49 @@
 
 namespace OpenApiGenerator;
 
+use JetBrains\PhpStorm\Pure;
 use OpenApiGenerator\Attributes\Controller;
 use OpenApiGenerator\Attributes\Info;
 use OpenApiGenerator\Attributes\Schema;
+use OpenApiGenerator\Attributes\SchemaSecurity;
+use OpenApiGenerator\Attributes\Server;
 use ReflectionAttribute;
 use ReflectionClass;
 use ReflectionException;
 
 class Generator
 {
-    private const OPENAPI_VERSION = "3.0.0";
+    public const OPENAPI_VERSION = "3.0.0";
+
+    /**
+     * default definition json schema.
+     *
+     * @var array
+     */
+    private array $definition = [
+        'info' => [],
+        'paths' => [],
+        'servers' => [],
+        'components' => [
+            'schemas' => [],
+            'securitySchemes' => [],
+        ],
+    ];
+
+    public function __construct(
+        private GeneratorHttp $generatorHttp,
+        private GeneratorSchemas $generatorSchemas,
+    ) {
+    }
+
+    /**
+     * @return Generator
+     */
+    #[Pure]
+    public static function factory(): Generator
+    {
+        return new self(new GeneratorHttp, new GeneratorSchemas);
+    }
```

Comment:
The constructor should be public to test it - avoiding reflections which are a pain - and the factory method should be an actual factory class. Soon, I'd like it to generate the scheme in json and yaml so the factory will be ready

Reply:
ok, i rename this method to `create`

Reply:
Last comment, could you just add parenthesis at the end of `GeneratorHttp` and  `GeneratorSchemas` please ? (PSR12)

Reply:
done